WireDomain/Sources/WireDomain/Notifications/PullEventsService.swift

```diff
@@ -0,0 +1,77 @@
+//
+// Wire
+// Copyright (C) 2025 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import Foundation
+import WireAPI
+
+// sourcery: AutoMockable
+protocol PullEventsServiceProtocol {
+    func startSync(
+        newEventID id: UUID
+    ) async throws
+}
+
+struct PullEventsService: PullEventsServiceProtocol {
+    private let userClientsLocalStore: any UserClientsLocalStoreProtocol
+    private let updateEventsLocalStore: any UpdateEventsLocalStoreProtocol
+    private let pendingEventsSync: any PullPendingUpdateEventsSyncProtocol
+    private let generateNotificationProvider: any GenerateNotificationProvider
+
+    enum Failure: Error {
+        case unableToPullPendingEvents(Error)
+    }
+
+    init(
+        userClientsLocalStore: any UserClientsLocalStoreProtocol,
+        updateEventsLocalStore: any UpdateEventsLocalStoreProtocol,
+        pendingEventsSync: any PullPendingUpdateEventsSyncProtocol,
+        generateNotificationProvider: any GenerateNotificationProvider
+    ) {
+        self.userClientsLocalStore = userClientsLocalStore
+        self.updateEventsLocalStore = updateEventsLocalStore
+        self.pendingEventsSync = pendingEventsSync
+        self.generateNotificationProvider = generateNotificationProvider
+    }
+
+    /// Starts syncing events with backend.
+    /// - parameter newEventID: The notification event id.
+    func startSync(
+        newEventID id: UUID
+    ) async throws {
+
+        let lastEventID = updateEventsLocalStore.lastEventID()
+
+        if lastEventID == nil {
+            updateEventsLocalStore.storeLastEventID(id: id)
```

Comment:
issue: Why do we store the last event here? Setting the received event id from the payload would mean we actually fetch all events after it, which would cause us to skip the event.